pkg/skaffold/build/builder_mux_test.go

```diff
@@ -92,6 +92,83 @@ func TestNewBuilderMux(t *testing.T) {
 	}
 }
 
+func TestGetConcurrency(t *testing.T) {
+	tests := []struct {
+		description         string
+		pbs                 []PipelineBuilder
+		cliConcurrency      int
+		expectedConcurrency int
+	}{
+		{
+			description: "default concurrency - builder and cli concurrency unset.",
+			pbs: []PipelineBuilder{
+				&mockPipelineBuilder{nil, "local"},
+				&mockPipelineBuilder{nil, "gcb"},
+			},
+			cliConcurrency:      -1,
+			expectedConcurrency: 1,
+		},
+		{
+			description: "builder concurrency set to less cli concurrency",
+			pbs: []PipelineBuilder{
+				&mockPipelineBuilder{util.IntPtr(1), "local"},
+				&mockPipelineBuilder{util.IntPtr(1), "local"},
+				&mockPipelineBuilder{nil, "gcb"},
+			},
+			cliConcurrency:      2,
+			expectedConcurrency: 2,
+		},
+		{
+			description: "builder concurrency set",
+			pbs: []PipelineBuilder{
+				&mockPipelineBuilder{util.IntPtr(2), "local"},
+				&mockPipelineBuilder{util.IntPtr(2), "local"},
+				// As per docs https://github.com/GoogleContainerTools/skaffold/blob/dbd18994955f5805e80c6354ed0fd424ec4d987b/pkg/skaffold/schema/v2beta26/config.go#L287
+				// nil concurrency defaults to 1
+				&mockPipelineBuilder{nil, "gcb"},
+			},
+			cliConcurrency:      -1,
+			expectedConcurrency: 1,
+		},
+		{
+			description: "builder concurrency set to 0 and cli concurrency set to 1",
+			pbs: []PipelineBuilder{
+				// build all in parallel
+				&mockPipelineBuilder{util.IntPtr(0), "local"},
+				&mockPipelineBuilder{util.IntPtr(0), "gcb"},
+			},
+			cliConcurrency:      1,
+			expectedConcurrency: 1,
+		},
+		{
+			description: "builder concurrency set to 0 and cli concurrency unset",
+			pbs: []PipelineBuilder{
+				// build all in parallel
+				&mockPipelineBuilder{util.IntPtr(0), "local"},
+				&mockPipelineBuilder{util.IntPtr(0), "gcb"},
+			},
+			cliConcurrency:      -1,
+			expectedConcurrency: 0,
+		},
+		{
+			description: "min non-zero concurrency",
+			pbs: []PipelineBuilder{
+				&mockPipelineBuilder{util.IntPtr(0), "local"},
+				&mockPipelineBuilder{util.IntPtr(3), "gcb"},
+				&mockPipelineBuilder{util.IntPtr(2), "gcb"},
+			},
+			cliConcurrency:      -1,
+			expectedConcurrency: 2,
+		},
```

Comment:
I think you need a test for all-GCB.  I suspec the code will return 1 but it should return 0 for this case.

Reply:
Let me add one.

Reply:
So, in case of GCB, it not possible to have nil Concurrency. 
See GCB build definition.  The `Concurrency` is an actual int.

```
type GoogleCloudBuild struct {
	// ProjectID is the ID of your Cloud Platform Project.
	// If it is not provided, Skaffold will guess it from the image name.
	// For example, given the artifact image name `gcr.io/myproject/image`, Skaffold
	// will use the `myproject` GCP project.
	ProjectID string `yaml:"projectId,omitempty"`
....

	// Concurrency is how many artifacts can be built concurrently. 0 means "no-limit".
	// Defaults to `0`.
	Concurrency int `yaml:"concurrency,omitempty"`

}


```

Reply:
@tejal29 should we still add a test case for it ?

Reply:
done.